lua/chatcommands/duel.lua

```diff
@@ -0,0 +1,284 @@
+                        requestDuel(player1, player2, points, kills)
+                    else
+                        player1:ChatPrint("[red]A duel must be worth a minimum of 1000 points.")
+                    end
+                else
+                    player1:ChatPrint("[red]The win condition can not be less than 1 kill.")
+                end
+            else
+                player1:ChatPrint("[red]The win condition can not be more than 20 kills.")
+            end
+        else
+            player1:ChatPrint("[red]Player " .. string.Implode(" ", args) .. " not found.")
+        end
+    end
+end)
+
+function hasPendingDuel(player1ID, player2ID, points)
+    local player1 = Player(player1ID)
+
+    for k, v in pairs(ActiveDuels) do
+        if (v.player2 == player2ID or v.player1 == player1ID or v.player2 == player1ID or v.player1 == player2ID) then
+
+            if (v.player2 == player2ID and v.player1 == player1ID or v.player2 == player1ID and v.player1 == player2ID) then
+                player1:ChatPrint("[red]You are currently in a duel with this player.")
+            else
+                if (v.player1 == player1ID) then
+                    player1:ChatPrint("[red]You are currently in a duel with [orange]" .. Player(v.player2):Nick() .. ".")
+                elseif (v.player2 == player1ID) then
+                    player1:ChatPrint("[red]You are currently in a duel with [orange]" .. Player(v.player1):Nick() .. ".")
+                else
+                    player1:ChatPrint("[red]Player is currently in a duel.")
+                end
+            end
+            return true
+        end
+    end
+
+    for k, v in pairs(DuelRequests) do
+        if (v.player2 == player2ID and v.points == points) then
+            player1:ChatPrint("[red]Player already has a duel request for that amount.")
+            return true
+        elseif (v.player1 == player1ID) then
+            player1:ChatPrint("[red]You already made a duel request.")
+            return true
+        end
+    end
+
+    for k, v in pairs(AcceptedDuels) do
+        if (v.player2 == player2ID and v.points == points) then
+            player1:ChatPrint("[red]Player already has a duel request for that amount.")
+            return true
+        elseif (v.player1 == player1ID) then
+            player1:ChatPrint("[red]You already made a duel request.")
+            return true
+        end
+    end
+
+    return false
+end
+
+
+function hasAcceptedDuel(playerID)
+    for k, v in pairs(ActiveDuels) do
+        if (v.player1 == playerID or v.player2 == playerID) then
+            return true
+        end
+    end
+
+    for k, v in pairs(AcceptedDuels) do
+        if (v.player1 == playerID or v.player2 == playerID) then
+            return true
+        end
+    end
+
+    return false
+end
+
+function requestDuel(player1, player2, points, kills)
+    if (not hasPendingDuel(player1:UserID(), player2:UserID(), points)) then
```

Comment:
Why bother with these UserIDs? Just use the actual player object

Reply:
Oh whoops, that is a leftover from me testing, something was acting up when I was testing with multiple clients running under the same steam account. I'll clean it up later

Reply:
There that should be a bit better